MdeModulePkg/Library/UefiBootManagerLib/BmBootDescription.c

```diff
@@ -888,7 +906,13 @@ BmMakeBootOptionDescriptionUnique (
   }
 
   Visited = AllocateZeroPool (sizeof (BOOLEAN) * BootOptionCount);

```

Comment:
This allocation could go before the loop and save the trouble of executing it if the allocation fails.

Not necessary to change though. Feel free to resolve.